### PR TITLE
Handle input prompts with Click

### LINKED ISSUE #262: Separate interactive prompts from logging
There are places in our code where interactive prompts and logging are mixed which is not great. See the `data_handling` module for examples.

That's where a proper CLI library such as `click` or `typer` would shine, since you can explicitly dissociate what's interactive from what's logging.

For now, all interactive message must be set to `warning` otherwise it won't be displayed at the default verbosity level.

### PR DESCRIPTION
**Pull request by ghisvail**
Closes #262

### CODE REVIEW

**Review by omar-rifai — CHANGES_REQUESTED** [review]
Thanks. This seems OK but why are you only handling this specific case? Other cases such as the interactive prompt in check_volume_location_in_world_coordinate_system seem to be require the same modification.

File: clinica/iotools/utils/data_handling.py [hunk 1]
```diff
@@ -1060,9 +1062,7 @@ def check_volume_location_in_world_coordinate_system(
                 "%-" + str(file_width) + "s%-" + str(center_width) + "s%-25.2f\n"
             ) % (basename(file), str(center), l2)
 
-        cmd_line = (
-            f"`clinica iotools center-nifti {abspath(bids_dir)} {abspath(bids_dir)}_centered --modality {modality}`"
-        )
+        cmd_line = f"`clinica iotools center-nifti {abspath(bids_dir)} {abspath(bids_dir)}_centered --modality {modality}`"
```
Comment (ghisvail) [inline clinica/iotools/utils/data_handling.py:1065]: Black reformatting

File: clinica/iotools/utils/data_handling.py [hunk 2]
```diff
@@ -85,7 +85,10 @@ def create_merge_file(
         ]
         row_participant_df.reset_index(inplace=True, drop=True)
         if len(row_participant_df) == 0:
-            cprint(msg=f"Participant {subject} does not exist in participants.tsv", lvl="warning")
+            cprint(
+                msg=f"Participant {subject} does not exist in participants.tsv",
+                lvl="warning",
+            )
```
Comment (ghisvail) [inline clinica/iotools/utils/data_handling.py:91]: Black reformatting

**Comment by ghisvail** [comment]
> Thanks. This seems OK but why are you only handling this specific case? Other cases such as the interactive prompt in check_volume_location_in_world_coordinate_system seem to be require the same modification.

Done.

**Comment by omar-rifai** [comment]
There are at least two additional cases in engine.py but I may have missed others.

**Comment by ghisvail** [comment]
They are hard to find. If you know where they are, please give me a list 👍

**Comment by omar-rifai** [comment]
I agree. I just ran a search with the term "input()" and "stdin" and the only two remaining seem to be in nifd_to_bids.py. That should cover all the cases.

File: clinica/pipelines/engine.py [hunk 3]
```diff
@@ -600,34 +584,19 @@ def update_parallelize_info(self, plugin_args):
...
-            # If plugin_args is None, create the dictionary
-            # If it already a dict, just update (or create -it is the same
-            # code-) the correct key / value
+            n_procs = click.prompt(
+                text="How many threads do you want to use?",
+                default=max(1, n_cpu - 1),
+                show_default=True,
+            )
+            click.echo(
+                f"Number of threads set to {n_procs}. You may set the --n_procs argument "
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply clinica/pipelines/engine.py:593]: There is no such feature in Click. We are either interactive or we are not.

[the remark this answers is not in the document]
Comment (ghisvail) [reply clinica/pipelines/engine.py:593]: Relying on interactivity + timeouts for cluster usage is very bad practice and should not be encouraged.

[the remark this answers is not in the document]
Comment (omar-rifai) [reply clinica/pipelines/engine.py:593]: Perhaps this issue is no longer relevant after [PR210](https://github.com/aramis-lab/clinica/pull/210), can you confirm it was correctly integrated with click as well?

[the remark this answers is not in the document]
Comment (ghisvail) [reply clinica/pipelines/engine.py:593]: We have got support for the `yes` flag, which should propagate to the `skip_question` parameter.

That being said, a better UX design would be a `--interactive/--no-interactive` flag which explicitly signals whether the user want to run the pipeline with or without prompts. `n_procs` can be set to an arbitrary value with or without interactivity and `yes` is a poor choice to express that in my opinion.

[the remark this answers is not in the document]
Comment (omar-rifai) [reply clinica/pipelines/engine.py:593]: Ok ! Can you please add this as enhancement issue? I'll merge this PR in the meantime.

**Review by omar-rifai — APPROVED** [review]

File: clinica/pipelines/engine.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ghisvail) [reply clinica/pipelines/engine.py:593]: See https://github.com/aramis-lab/clinica/issues/306

**Pull request merged**